tests/rptest/tests/cluster_quota_test.py

```diff
@@ -215,6 +215,30 @@ def fetch(self, consumer, messages_amount, timeout_sec=300):
             if now > deadline:
                 raise TimeoutError()
 
+    def make_producer(self, client_id: Optional[str], *args,
```

Comment:
Isn't `Optional[str] == str | None`? Or is it a convention/preference to use the former for optional parameters and the latter for `None`'able required arguments?

I'm adding `None` as a default value now, but I'm curious because I haven't seen these being differentiated.

Reply:
Yeah, they are equivalent afaik. I just think the language is a little unclear, viz an "optional" param which is not really optional _at the call site_. `None`able, but you would have to pass the `None` explicitly. 

Anyway, typing module's fault; not worth dwelling on, just yelling at clouds 🙂